nilearn/surface/surface.py

```diff
@@ -595,20 +674,14 @@ def load_surf_mesh(surf_mesh):
                 surf_mesh.endswith('inflated')):
             coords, faces = nibabel.freesurfer.io.read_geometry(surf_mesh)
         elif surf_mesh.endswith('gii'):
-            try:
-                coords = gifti.read(surf_mesh).getArraysFromIntent(
-                    nibabel.nifti1.intent_codes[
-                        'NIFTI_INTENT_POINTSET'])[0].data
-            except IndexError:
-                raise ValueError('Gifti file needs to contain a data array '
-                                 'with intent NIFTI_INTENT_POINTSET')
-            try:
-                faces = gifti.read(surf_mesh).getArraysFromIntent(
-                    nibabel.nifti1.intent_codes[
-                        'NIFTI_INTENT_TRIANGLE'])[0].data
-            except IndexError:
-                raise ValueError('Gifti file needs to contain a data array '
-                                 'with intent NIFTI_INTENT_TRIANGLE')
+            if LooseVersion(nibabel.__version__) >= LooseVersion('2.1.0'):
+                gifti_img = nibabel.load(surf_mesh)
+            else:
+                gifti_img = gifti.read(surf_mesh)
+            coords, faces = _load_surf_mesh_img_to_data(gifti_img)
+        elif surf_mesh.endswith('.gii.gz'):
+            gifti_img = _load_surf_files_gifti_gzip(surf_mesh)
+            coords, faces = _load_surf_mesh_img_to_data(gifti_img)
         else:
             raise ValueError(('The input type is not recognized. %r was given '
                               'while valid inputs are one of the following '
```

Comment:
below, maybe we could
```
try:
    coords, faces = surf_mesh
except Exception:
#...
```
rather than checking if it is a list with `elif isinstance(surf_mesh, list)`; for example it would seem perfectly natural for a user to give coords, faces as a 2-tuple.

sorry this is not directly related to shipping fsaverage5 files but since you're improving this function this could be fixed as well

Reply:
Do you wanna add ```collections.Iterable``` here rather than ```elif isinstance(surf_mesh, list)``` ?

Reply:
yes, in this case surf_mesh should be any iterable with exactly two elements.

Reply:
```dict()``` is an iterable ?

Reply:
Can you see the recent commit ?

Reply:
yes. I'm very sorry for the noise, please ignore my whole comment. maybe just allow tuple as well as list, as this will cover use cases such as people writing `mesh = coords, faces`, or representing meshes with namedtuples

Reply:
> dict() is an iterable ?

yes, if you prefer collections.Sequence is more precise

Reply:
> Can you see the recent commit ?

yes, it's great! and CI is green again, this is awesome!